core/utils/dom.js

```diff
@@ -271,3 +278,48 @@ Blockly.utils.dom.getTextWidth = function(textElement) {
   }
   return width;
 };
+
+/**
+ * Gets the width of a text element using a faster method than `getTextWidth`.
+ * This method requires that we know the text element's font family and size in
+ * advance. Similar to `getTextWidth`, we cache the width we compute.
+ * @param {!Element} textElement An SVG 'text' element.
+ * @param {string} fontSize The font size to use.
+ * @param {string} fontFamily The font family to use.
+ * @return {number} Width of element.
+ */
+Blockly.utils.dom.getFastTextWidth = function(textElement,
+    fontSize, fontFamily) {
+  var text = textElement.textContent;
+  var key = text + '\n' + textElement.className.baseVal;
+  var width;
+
+  // Return the cached width if it exists.
+  if (Blockly.utils.dom.cacheWidths_) {
+    width = Blockly.utils.dom.cacheWidths_[key];
+    if (width) {
+      return width;
+    }
+  }
+
+  if (!Blockly.utils.dom.canvasContext_) {
+    // Inject the canvas element used for computing text widths.
+    var computeCanvas = document.createElement('canvas');
+    computeCanvas.className = 'blocklyComputeCanvas';
+    document.body.appendChild(computeCanvas);
+
+    // Initialize the HTML canvas context and set the font.
+    // The context font must match blocklyText's fontsize and font-family
+    // set in CSS.
+    Blockly.utils.dom.canvasContext_ = computeCanvas.getContext('2d');
+    Blockly.utils.dom.canvasContext_.font = fontSize + 'pt ' + fontFamily;
```

Comment:
Don't you need to set the fontsize and fontfamily each time this is called since those are being passed in to the method?

Reply:
Oh yes thats true. I'll update that and run some quick tests to make sure it is still faster :)